lib/webauthn/public_key_credential/request_options.rb

```diff
@@ -14,6 +14,10 @@ def initialize(rp_id: nil, allow_credentials: nil, allow: nil, user_verification
         @allow_credentials = allow_credentials
         @allow = allow
         @user_verification = user_verification
+
+        if relying_party.legacy_u2f_appid && @extensions&.[](:appid).nil? && @extensions&.[]("appid").nil?
+          @extensions = (@extensions || {}).merge(appid: relying_party.legacy_u2f_appid)
```

Comment:
Loved the idea! For sure we both agree that is difficult to read. However, the only problem here is that when the `extensions` that came in the params are something like `{ "appid" => 'https://example.com' }` we will end up having two `appid` values in the `extensions` attribute.

Maybe is a good time to think about normalizing the keys of the `extension` hash?

Reply:
Well, maybe but not too worried about string keys because this is the user typing this option for options for get, not the params coming from the browser.